src/test/java/CompatibilityTest.java

```diff
@@ -31,6 +33,32 @@ public void v2() throws Exception {
         assertFieldType(i, String[].class, "y");
         assertFieldType(i, Object.class, "z");
     }
+    
+    @Test
+    public void testClassesAreOnlyReWrittenWhenNeeded() throws Exception {
+        System.out.println("rewrtting tests");
+        final Transformer t = new Transformer();
+        AntClassLoader cl = new AntClassLoader() {
+            @Override
+            protected Class<?> defineClassFromData(File container, byte[] classData, String classname) throws IOException {
+                byte[] rewritten = t.transform(classname, classData);
+                if (rewritten!=classData ) {
+                    fail(classname + " was rewritten without need");
+                }
+                return super.defineClassFromData(container, rewritten, classname);
+            }
+        };
+
+        cl.addPathComponent(new File("target/test-classes/v2"));
```

Comment:
These are not on the classpath - so you can't load them with a class loader.
You can't move them to the classpath and use loadClass as you require **multiple versions of the same class** for these tests.

So if you require a nice safe way then you actually need to move them all to a sub folder with an identification resource that can be loaded to find the location - but even that is potentially doomed to fail.

Right now this test is fine in Eclipse and the command line at the root of the project.  If it breaks for other IDEs it was already broken so I have not broken anything.

@tfennelly Yes ^^^